### PR TITLE
feat: convert `patient slice getters` to classes

### LINKED ISSUE #561: feat: convert `patient_slice_getters` to classes
* Shared `protocol` to make modularity easier
* Separates loading and init, to make for faster config validation

### PR DESCRIPTION
**Pull request by MartinBernstorff**
Depends on #563, review that first.

Fixes #561.

* Shared protocol to make composition much easier
* Separates loading and init, to make for faster config validation

### CODE REVIEW

File: psycop/common/feature_generation/sequences/patient_loaders.py [hunk 1]
```diff
@@ -13,13 +13,23 @@
...
+
 
 class EventDfLoader(ABC):
     @abstractmethod
     def load_events(self) -> pl.LazyFrame:
         ...
 
 
+@Registry.utilities.register("list_creator")
+def list_creator(*args: T) -> list[T]:
```
Comment (KennethEnevoldsen) [inline psycop/common/feature_generation/sequences/patient_loaders.py:28]: So here we just say that the type should be the same? (I am not sure we want to say that)

File: psycop/projects/t2d/feature_generation/cohort_definition/t2d_cohort_definer.py [hunk 2]
```diff
@@ -21,9 +21,12 @@
     get_first_diabetes_indicator,
 )
 
+from .....common.sequence_models.registry import Registry
```
Comment (KennethEnevoldsen) [inline psycop/projects/t2d/feature_generation/cohort_definition/t2d_cohort_definer.py:24]:
```suggestion
from psycop.common.sequence_models.registry import Registry
```

**Review by KennethEnevoldsen — APPROVED** [review]
Looks good - let us just discuss the type but otherwise it looks fine

File: psycop/common/feature_generation/sequences/patient_loaders.py (hunk 1 above)
Comment (MartinBernstorff) [reply psycop/common/feature_generation/sequences/patient_loaders.py:28]: If the input is `list_creator(1, "2")`, the type of `*args` is `int | str`. I think the return type should match that :-)

Comment (MartinBernstorff) [reply psycop/common/feature_generation/sequences/patient_loaders.py:28]: (Have refactored in dependent PR to move it to `registry.py`, deleting here)

**Comment by MartinBernstorff** [comment]
Waiting for dependent branch, otherwise ready to merge.

**Pull request closed without merging**